app/views/dishes/show.html.erb

```diff
@@ -0,0 +1,18 @@
+<h1><%= @dish.name %></h1>
+<p><%= @dish.description %></p>
+
+<h3>Ingredients</h3>
+<p><u>Name -- Calories</p></u>
+<% @dish.ingredients.each do |ingredient| %>
+  <%= ingredient.name %>
+  <%= ingredient.calories %> <br>
+<% end %>
+
+<h4>Total Calories in This Meal: <%= "#{@dish.total_calories}" %></h4>
+
+<h3>Think it needs something more?</h3>
+<%= form_with url:"/dishes/#{@dish.id}", method: :patch, local: true do |f| %>
+  <%= f.label "Ingredient ID" %>
+  <%= f.number_field :add_ingredient %>
+  <%= f.submit "Submit" %>
+<% end %>
```

Comment:
While this form is what I would consider RESTful, it is not RESTful in what it's aiming to accomplish. The task we are given is to associate a new ingredient to a dish, this does not happen on the dish resource, this is a new relationship being CREATED on the DISH_INGREDIENTS table between a dish and an ingredient. We are not updating anything on the dish table itself, therefore a `PATCH` to "/dishes/#{@dish.id}" is incorrect.

Therefore, your form should be making a `POST` request to `"/dishes/#{@dish.id}/ingredients"`.

```erb
<%= form_with url:"/dishes/#{@dish.id}/ingredients", method: :post, local: true do |f| %>
  <%= f.label "Ingredient ID" %>
  <%= f.number_field :add_ingredient %>
  <%= f.submit "Submit" %>
<% end %>
```

Could I also suggest changing `:add_ingredient` to `:ingredient_id` 😅